tests/test_cases/truncate.c

```diff
@@ -0,0 +1,72 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <fcntl.h>
+#include <sys/stat.h>
+#include <unistd.h>
+
+const char *FILENAME = "testfile.txt";
+const off_t OFFSET = 123;
+
+void check_filesize(off_t target_size)
+{
+	struct stat st;
+	stat(FILENAME, &st);
+	if (st.st_size != target_size)
+	{
+		fprintf(
+			stderr,
+			"ERROR: EXPECT FILESIZE TO BE %d BUT GOT %d\n",
+			target_size,
+			st.st_size
+		);
+		exit(1);
+	}
+	printf("Success\n");
+}
+
+void check_success(int r)
+{
+	if (r == -1)
+	{
+		perror("ERROR WHEN TRUNCATING FILE");
+		exit(1);
+	}
+}
+
+
+int main(int argc, char **argv)
+{
+	int fd;
+
+	struct stat st;
+	stat(FILENAME, &st);
+	off_t filesize = st.st_size;
+	off_t filesize_inc = filesize + OFFSET;
+	
+
+	printf("Test truncate(const char *, off_t)\n");
```

Comment:
This should be deterministic because its single process, but its still worth noting that for these tests you generally have to fflush after prints. It something were looking into fixing but theres no guarantee of an immediate print from Lind right now without it.

Reply:
Done